signals/signal_exporter.py

```diff
@@ -0,0 +1,483 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Any, Dict, List, Optional, Tuple
+import redis
+import platform
+import json
+import time
+import uuid
+import logging
+
+
+"""
+State/Event Signal Module
+
+Adds two new, simple-to-use objects:
+ - SignalExporter      (for publishing state signals and handling subscribers + responses)
+ - SignalResponder     (for receiving state signals, locking onto publishers, and publishing responses)
+
+Also provides two dataclass specifications:
+ - Signal              (state signal protocol definition)
+ - Response            (response protocol definition)
+
+Combining redis pubsub features with state signal + response protocols, 
+these additions make state signal publishing, subscribing, receiving, 
+and responding incredibly easy to integrate into any code.
+"""
+
+
+def _create_logger(
+    class_name: str, process_name: str, log_level: str
+) -> logging.Logger:
+    """
+    Creates and returns logging.Logger object for detailed logging.
+    Used by SignalExporter and SignalResponder.
+    """
+    logger = logging.getLogger(class_name).getChild(process_name)
```

Comment:
Do you think we could modify this to use a base logger with a specific name for the signals? As well as use a global log-level for the loggers that makes the messages easy to control? Something like:

```python

def setup_logging():
    base_logger = logging.getLogger("state-signals")
    stream_handler = logging.StreamHandler()
    try:
        base_logger.setLevel(log_level)
        stream_handler.setLevel(log_level)
     except ValueError:
          ....
        
    formatter = ....
    stream_handler.setFormatter(formatter)
    base_logger.addHandler(stream_handler)
    
    return base_logger

_base_logger = _get_base_logger()

def _create_logger(class_name: str, process_name: str, log_level: str) -> logging.Logger:
    """..."""

    return _base_logger.getChild(f"{class_name}.{process_name}")
```

Or even just:

```python
def _create_logger(...):
    logger = logging.getLogger("state-signals").getChild(class_name).getChild(process_name)
    ...
```

This would establish the standard practice logging hierarchy as well as make these log messages easy to control from other apps. For instance, I could mute signal logs in my app and redirect to a file with:

```python
import logging
import state_signals

logging.getLogger("state-signals").setLevel(logging.ERROR)
logging.getLogger("state-signals").addHandler(logging.FileHandler("/tmp/state-signals.log"))
```

Or whatever other sort of logging configuration my app needs.